<a href="https://colab.research.google.com/github/wgalindo1453/PythonTeachingMaterial/blob/main/variablesTeachingScript.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##JupyterVox Teaching Script
#Lesson 3, 03/25/23




##General Notes to Instructors


1.   Take notes on the difficulties/issues encountered by the students
2.  Be patient and wait for the student to finish, don’t jump in to assist immediately. Tell the student to ask for help when the student feels help is needed. We were intrusive in the first two sessions. If you feel you will need to type on the student’s keyboard, please ask permission from the student first. Also
narrate what it is you are doing/looking for/trying to understand.
1. When you move from one task to another (or to a different subtask), communicate this to the student. They are highly aural. As you communicate which task or subtask you are starting, it will help them to understand where they are at in terms of their present location relative to the entirety of what they have to accomplish for the morning   
2. At the end of the session, we will need to ask the student for feedback for teaching improvements.
  * We may also ask the parents for feedback
1.We will separate the students to keep them focused on coding
2.Need to make sure students don’t stray too much from teaching materials
1. Ask Dr. Ewoldt and Dr. Neely for help, if we can’t manage it
2. Give the students an overview of today’s tasks at the beginning.
Need to disable auto-complete, auto-close brackets, and diagnose
1.  Need to disable auto-complete, auto-close brackets, and diagnose (and all editing assistants) in Colab settings.
2.   Take notes on the difficulties/issues encountered by the students


##Task 0: Set-up & Overview of today’s tasks   

1.   Explain new layout as students enter
1.   Have students set up their laptops & braille readers on individual table then go to community table for snack/drink/fellowship
2.  Dr. Ewoldt will explain overview of tasksp
2.  Need to disable auto-complete, auto-close brackets, and diagnose (and all editing assistants) in Colab settings.




##Task 1: Recap of last week (20min)





### Notes to instructor
  1. Tell the student we will not assist, and they will need to work on this themselves

### Lecture
1. Create a new Colab notebook
2. Create a new code cell, code “3.3+5.3” (sum, decimal), execute,and examine the results.

In [ ]:
3.3 + 5.3

8.6

3. Create a new code cell, code “7-4*12” (complex), and execute:

In [ ]:
# This is the code format
7-4*12

-41

4. Create a new code cell, code “(10+12)/(3-1)” (more parentheses), and execute

In [ ]:
(10+12)/(3-1)

11.0

5. Create a new code cell, code “(5-(3-2))**10” (nested parentheses and exponent), and execute. This one probably requires starting with (3-2), then to 5-(3-2), then to the final equation.

In [ ]:
(5-(3-2))**10

1048576

6. Move the cursor back to the third “parentheses” cell and change the sum to “(9-3) *(7.2+4)”.

In [ ]:
(9-3) *(7.2+4)

67.19999999999999

#Task 2: Variables



## Notes to instructor:

1. we will cover assignments, variable computation, identifier, variable types, and type conversion, and keywords
2. The students will encounter errors in this task. Let the student read the errors first to see how they react.
1. This task is long, will need to take a break

## Lecture

### Variable assignments (10min)
1. Teach the concept of variables
  *  A variable is a symbol to represent a math object. A variable is usually used to store some value/number
  *  “=” is used to do assignments, I.e., assign values to a variable. Values can be changed or reassigned.
  * A variable can be used in place of numbers in math computation
  * Variables can be reassigned


### Math with Variables
1. Let the student write the following code in a code cell

In [ ]:
a = 12
print(a)

12


2. Code in a new cell

In [ ]:
b = a + 13.2 # ask the student what should be the result
print(b)

25.2


3. Code in a new cell

In [ ]:
a = 15
b = a + 13.2 # ask the student what b would be now
print(b)

28.2


#### Practice


  >a. Ask the student to create a variable with the name “x” and value   12.5. Then do the math to add another 12.5 to x.

>b. Ask the student to compute 45/15, with 15 being stored in a variable.

### Variable names (identifier) (15min)

1.  Teach the rule of variable name
  *   Must start with a letter
  *   Can only have letters, digits, and und. erscore “_”
  *   Teach the rule of variable name

1.   Try assignments to different names
  *  Note bad names will cause errors, let the student read the errors
  *  Use the following names to do assignments, e.g., “a1 = 333”.
     >1. X1
     >2. x1
     >3. firstName
     >4. last_name
     >5. ZipCode1
     >6. 1stZipCode
     >7. name+
     >8. email@work

1.  Practices
 * Number type, string, boolean (true false)
 * There are other types, we will learn in the future
 * The main reason of using type is because some types cannot be mixed together


### Variable types (30 minutes)

1.   Tell the students variable have types
    * Number type, string, boolean (true false)
    * There are other types, we will learn in the future
    * There are other types, we will learn in the future


#### Integer type
  * Variable is used to hold whole numbers
  * E.g., “a=12”
  * In a new code cell, type

In [ ]:
a = 12
type(a)

int

#### Float type
  * Variable is used to hold decimal numbers; float is a special name in Computer Science for decimal numbers
 * E.g., “b = 12.3”
 * In a new code cell, type,


In [ ]:
b = 12.3
type(b)

float

#### String type
  * Variable is used to hold a string of characters/text. Text must be quoted (single or double quote marks).
  * E.g., “s = “i am a string variable””
  * In a new code cell, type

In [ ]:
  s = "i am a string variable"
  type(s)

str

 * In a new code cell, type:

In [ ]:
s1 = "my name is "
s2 = "John" # the student’s name
s3 = s1+s2 # here “+” is used to concatenate strings
print(s3)

my name is John


#### Boolean

  *Variable is used to hold a True or False value. Must be uppercase T and F. Typically used to make decision, e.g., do something when a variable is True

* E.g., “m = True”
* In a new code cell, type

In [ ]:
m = True
type(m)

bool

In [ ]:
n = False
type(n)

bool

    * Tell the students there are more types, and we will learn them when they are used

### Mixing types in equations
  * Tell the student different types of variables cannot always be used in the same equation

#### Ints and Floats
Int and float variables can be used in the same equation. For example, in a new cell

In [ ]:
x = 12
y = 13.3
z = x + y
print(z)

25.3


#### Strings and Ints/Floats

String and int/float variables cannot be used in the same equation. For example, in a new code cell,

In [ ]:
s = "a string" # let the student type anything
x = 12
s3 = s + x
#There should be an error

TypeError: ignored

 The same holds for string and Boolean. For example

In [ ]:
s = "string" # let the student type anything
m = False
s3 = s + m
#There again should be an error

TypeError: ignored

### Booleans and Ints/Floats
However, boolean can be combined with integer/floats. Bascially, True is treated as 1, and False is treated as 0. So they are basically integers. For example

In [ ]:
m = False
x = 12
y = m+x
print(y)


12


#### Practices
  >1. Can a string be added to a string?
  >2. Can a string be added to an integer?
  >3. Can an integer be divided by a float?
  >4. Does False subtracts 25.3 work?

### Type conversion  
* Tell the students sometimes we can convert integers/floats/booleans from and to strings

#### Converting to strings.
Use function str(). In three new code cells,

In [ ]:
x = 12
s = "the value of x is " + str(10) # “+” now works
print(s)

the value of x is 10


In [ ]:
y = 12.3
s = "the value of y is " + str(y)
print(s)

the value of y is 12.3


In [ ]:
m = False
s = "the value of m is " + str(m)
print(s)

the value of m is False


#### Converting strings to Ints/Floats/Boolean

1. convert string to int

In [ ]:
s = "12"
x = int(s)
type(x)
print(x)

12


2. convert string to float

In [ ]:
s = "12.3"
y = float(s)
type(y)

float

3. convert string to boolean

In [ ]:
s = "True"
m = bool(s)
type(m)

bool

4. Ask the student: What happens to int(“12.3”)? Try it out
     

In [ ]:
int("12.3")

ValueError: ignored

#Task 3: A simple calculator



## Notes to instructor


1.   We will first show them how to use the input() function
2.   Ask the student to develop the code

## Lecture


### The input() function

1. Use the input() function. Tell the student input() can be used to obtain user input and store it in a variable
2. For example, in a new code cell
3. *Note that the student may have trouble finding the input box. We need to be patient and wait*

In [ ]:
print("Please input a string:")
s = input()
print(s)

Please input a string:
j
j


4. For a calculator, we need to accept inputs as numbers, change the code to




In [ ]:
print("Please input the first number:")
a = input()
print(a)

Please input the first number:
a
a


5. Ask the student what the type is “a,” and how to verify that. “a” should be string, and it can be verified with type(a)

In [ ]:
type(a)

str

5. Ask the student can we use a string to do math computation. The answer should be No.

6. Ask the student how to convert the string to numbers? The answer should be int() or float().  
For generality, we will use float(), so the code should bAsk the student how to convert the string to numbers? The answer should be int() or float().  
For generality, we will use float(), so the code should b

In [ ]:
print("Please input the first number:")
a = input()
x = float(a) # use a new variable x to hold the value

Please input the first number:
8.5


### The Calculator

7. For the 2nd variable, in the same cell, ask the student to figure out the following for the 2nd variable,  


In [ ]:
print("Please input the second number:")
b = input()
y = float(b)

Please input the second number:
9.0


8. At last, in the same cell, let the student add the code to do the math,


In [ ]:
z = x + y
print(z)

17.5


9. Ask the student to change the operator “+” to others

10. A more powerful equation calculator
  * We will use the eval function. It does automatically evaluation/calculation of equations.
  * Python provides this function to simply coding. However, as a programmer we should be able to implement this eval function ourselves
  * In a new code cell, type,

In [ ]:
print("Please input your math equation:")
eq = input()
result = eval(eq)
print(result)

Please input your math equation:
0
0


# Wrapup

1. Let the student share the Colab file with us. We need to send the files back to the parents

2. Ask for feedback from the students and parents

3. Additional questions for the students:  

  * Can you screen read pictures

  * How do you read math plots

  * Any math problem you want to learn how to solve with a computer?